tests/model/test_model.py

```diff
@@ -3744,24 +3744,33 @@ def test_get_next_unread_topic(
 
         assert unread_topic == next_unread_topic
 
-    def test_get_next_unread_pm(self, model):
-        model.unread_counts = {"unread_pms": {1: 1, 2: 1}}
-        return_value = model.get_next_unread_pm()
-        assert return_value == 1
-        assert model.last_unread_pm == 1
-
-    def test_get_next_unread_pm_again(self, model):
-        model.unread_counts = {"unread_pms": {1: 1, 2: 1}}
-        model.last_unread_pm = 1
-        return_value = model.get_next_unread_pm()
-        assert return_value == 2
-        assert model.last_unread_pm == 2
-
-    def test_get_next_unread_pm_no_unread(self, model):
-        model.unread_counts = {"unread_pms": {}}
-        return_value = model.get_next_unread_pm()
-        assert return_value is None
-        assert model.last_unread_pm is None
+    @pytest.mark.parametrize(
+        "unread_pms, last_unread_pm, next_unread_pm",
+        [
+            case(
+                {(1, "pm"), (2, "pm2")},
+                None,
+                (1, "pm"),
+                id="unread_present_after_no_state",
+            ),
+            case(
+                {(1, "pm"), (2, "pm2")},
+                (1, "pm"),
```

Comment:
Similarly in the test refactor here, this is setting model.last_unread_pm to a tuple, which is different.